src/python/pants/pantsd/process_manager.py

```diff
@@ -105,38 +106,47 @@ def _maybe_cast(item, caster):
       return item
 
   @classmethod
-  def _deadline_until(cls, closure, timeout, wait_interval=WAIT_INTERVAL_SEC):
+  def _deadline_until(cls, closure, action_msg, timeout=FAIL_WAIT_SEC,
+                      wait_interval=WAIT_INTERVAL_SEC, info_interval=INFO_INTERVAL_SEC):
     """Execute a function/closure repeatedly until a True condition or timeout is met.
 
     :param func closure: the function/closure to execute (should not block for long periods of time
                          and must return True on success).
+    :param str action_msg: a description of the action that is being executed, to be rendered as
+                           info while we wait, and as part of any rendered exception.
     :param float timeout: the maximum amount of time to wait for a true result from the closure in
                           seconds. N.B. this is timing based, so won't be exact if the runtime of
                           the closure exceeds the timeout.
     :param float wait_interval: the amount of time to sleep between closure invocations.
+    :param float info_interval: the amount of time to wait before and between info logging that
+                                we're still waiting for the `action_msg`.
```

Comment:
this param description doesn't parse ('that we're still waiting for the action_msg'?)